frontend/app/services/SubscriptionService.scala

```diff
@@ -131,12 +133,29 @@ object SubscriptionService {
   }
 }
 
-class SubscriptionService(val tierPlanRateIds: Map[ProductRatePlan, String],
-                          val zuoraSoapService: ZuoraSoapService,
+class SubscriptionService(val zuoraSoapService: ZuoraSoapService,
                           val zuoraRestService: ZuoraRestService) extends AmendSubscription with LazyLogging {
 
   import SubscriptionService._
-  val MembershipProductType = "Membership"
+  val membershipProductType = "Membership"
+  val productRatePlanChargeModel = "FlatFee"
+
+  def tierPlanRateId(productRatePlan: ProductRatePlan): Future[String] =
+    membershipProducts.map { products =>
+      val zuoraRatePlanId = for {
+        product <- products.find(_.`Tier__c`.contains(productRatePlan.salesforceTier))
+        zuoraRatePlan <- productRatePlan match {
+          case FriendTierPlan | StaffPlan => product.activeRatePlans.headOption
+          case paidTierPlan: PaidTierPlan =>
```

Comment:
Your remark follows my line of thoughts, but for some reason the compiler identifies this value as the base class, not as a `PaidTierPlan`. It'd be a good exercise to investigate why this happens.